src/lib/translators/mongo.ts

```diff
@@ -932,17 +932,25 @@ function transformPivot(step: Readonly<S.PivotStep>): MongoStep[] {
         },
       },
     },
-    /**
-     * Then we include back in every document created in the previous step the index columns
-     * (still accessible in the _id object)
-     */
-    { $addFields: addFieldsStep },
-    /**
-     * Then we replace the root of the documents tree to get our columns ready for
-     * our needed table-like, unnested format
-     */
-    { $replaceRoot: { newRoot: '$_vqbAppTmpObj' } },
   ];
+
+  if (Object.keys(addFieldsStep).length > 0) {
+    return [
+      ...partialRet,
+      /**
+       * Then we include the index columns back in every document created in the previous step
+       * (still accessible in the _id object)
+       */
+      { $addFields: addFieldsStep },
+      /**
+       * Then we replace the root of the documents tree to get our columns ready for
+       * our needed table-like, unnested format
+       */
+      { $replaceRoot: { newRoot: '$_vqbAppTmpObj' } },
+    ];
+  } else {
+    return [...partialRet, { $replaceRoot: { newRoot: '$_vqbAppTmpObj' } }];
+  }
```

Comment:
This is a bit complex, and leads to repetition.
I think it reads way easier if you just mutate the array that you're creating:

Also `partialRet` does not convoy a lot of meaning IMHO

```js
let pivotMongoAggStages = [...]

if (Object.keys(addFieldsStep).length > 0) {
      /**
       * Then we include the index columns back in every document created in the previous step
       * (still accessible in the _id object)
       */
      pivotMongoAggStages.push({ $addFields: addFieldsStep })
}
/**
       * Then we replace the root of the documents tree to get our columns ready for
       * our needed table-like, unnested format
       */
pivotMongoAggStages.push({ $replaceRoot: { newRoot: '$_vqbAppTmpObj' } });

return pivotMongoAggStages;
```

What do you think?

Reply:
Yes much better ! thx